src/techs/techs.service.ts

```diff
@@ -517,19 +518,37 @@ export class TechsService {
         }
     }
 
-    private async userCanChangeCategory(categoryId: number, user: UserReq) {
+    private async userCanChangeCategory(
```

Comment:
Should we move this to the ability folder with all the authorization stuff? 
Like this https://github.com/chingu-x/chingu-dashboard-be/blob/dev/src/ability/conditions/features.ability.ts

Reply:
OK

Reply:
So this would be in it's own file like categories.ability.ts?

Reply:
I would put under voyages, actually `voyage-teams.ability.ts` (already exist) sounds good

Reply:
I found one issue with moving userCanChangeCategory - it requires an instance of PrismaService to make a query.  voyage-teams.ability.ts doesn't have this, so I would have to create one or pass it as an argument to the function.  Is this worth doing?

Reply:
You can use `import prisma from "../../prisma/client";` like here https://github.com/chingu-x/chingu-dashboard-be/blob/efdc177d7f4eaae80c9eefcc272bf1479042d029/src/ability/conditions/features.ability.ts#L2

which is this https://github.com/chingu-x/chingu-dashboard-be/blob/dev/src/prisma/client.ts

The main thing is we don't want to create new prisma clients every time we need to use it, so we import this shared one.

Reply:
That makes sense - thanks!